framework/src/executioners/Eigenvalue.C

```diff
@@ -114,31 +224,160 @@ Eigenvalue::execute()
     _eigen_problem.petscOptionsInserted() = true;
   }
 #endif
+}
+
+void
+Eigenvalue::setFreeNonlinearPowerIterations(unsigned int free_power_iterations)
+{
+#if !PETSC_RELEASE_LESS_THAN(3, 12, 0)
```

Comment:
This is a situation where `#if` is appropriate because we care about the value of the macro, not simply about whether it exists

Reply:
It is really hard for me to get something right, haha